### PR TITLE
KAFKA-13821: Update Kafka Streams WordCount demo to new Processor API

### PR DESCRIPTION
**Pull request by tang7526**
https://issues.apache.org/jira/browse/KAFKA-13821

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by tang7526** [comment]
@chia7712 Could you help to review this PR? thanks.

File: streams/examples/src/main/java/org/apache/kafka/streams/examples/wordcount/WordCountTransformerDemo.java [hunk 1]
```diff
@@ -80,17 +83,17 @@ public void init(final ProcessorContext context) {
                                 final KeyValue<String, Integer> entry = iter.next();
 
                                 System.out.println("[" + entry.key + ", " + entry.value + "]");
-
-                                context.forward(entry.key, entry.value.toString());
+                                final Record<String, String> record = new Record<>(entry.key, entry.value.toString(), timestamp);
```
Comment (bbejeck) [inline streams/examples/src/main/java/org/apache/kafka/streams/examples/wordcount/WordCountTransformerDemo.java]: nit: we could just inline the `new Record(...)` in the `context.forward()`

**Review by bbejeck — COMMENTED** [review]
One minor comment, but otherwise LGTM

**Comment by bbejeck** [comment]
Thanks for the contribution @tang7526!

**Comment by bbejeck** [comment]
@vvcephei | @jeqo for a second pair of eyes

File: streams/examples/src/main/java/org/apache/kafka/streams/examples/wordcount/WordCountTransformerDemo.java (hunk 1 above)
Comment (tang7526) [reply streams/examples/src/main/java/org/apache/kafka/streams/examples/wordcount/WordCountTransformerDemo.java]: @bbejeck Done. thank you for your advice.

**Review by bbejeck — APPROVED** [review]
Thanks for the update @tang7526 , LGTM

**Review by jeqo — APPROVED** [review]
LGTM
thank you, @tang7526

**Comment by bbejeck** [comment]
Merged #12139 into trunk

**Comment by bbejeck** [comment]
Thanks, @tang7526 for the contribution!

**Pull request merged**